channels/offline-channel-js/src/OfflineChannel.ts

```diff
@@ -329,86 +344,181 @@ export class OfflineChannel extends BaseTelemetryPlugin implements IChannelContr
             }
 
             //flush only flush max batch size event, may still have events lefts
-            function _flushInMemoItems(unload?: boolean) {
+            // **********************************************************************************
+            // do you need to add function to flush each individual batch (for addevent process)
+            function _flushInMemoItems(unload?: boolean, mapKey?: number | EventPersistence) {
                 try {
                     // TODO: add while loop to flush everything
-                    let inMemo = _inMemoBatch;
-                    let evts = inMemo && inMemo.getItems();
-                    if (!evts || !evts.length) {
-                        return;
-                    }
-                    let payloadArr:string[] = [];
-                    let size = 0;
-                    let idx = -1;
-                    let criticalCnt = 0;
-                    arrForEach(evts, (evt, index) => {
-                        let curEvt = evt as IPostTransmissionTelemetryItem
-                        idx = index;
-                        let payload = _getPayload(curEvt);
-                        size += payload.length;
-                        if (size > _maxBatchSize) {
+                    let hasEvts = false;
+                    // can use objForEachKey(_inMemoMap, (key)), but keys returned by this function is always string
+                    arrForEach(PersistenceKeys, (key) => {
+                        let inMemoBatch: InMemoryBatch = _inMemoMap[key];
+                        let inMemo = inMemoBatch;
+                        let evts = inMemo && inMemo.getItems();
+                        if (!evts || !evts.length) {
                             return;
                         }
-                        if(curEvt.persistence == EventPersistence.Critical) {
-                            criticalCnt ++;
+                        if (_splitEvts && mapKey && mapKey !== key) {
+                            // if split evts is set to true
+                            // specific mapkey is defined
+                            // for key !== mapkey, only check if there are any events left in order to refresh timer
+                            hasEvts = !!evts.length;
```

Comment:
nit: "if" there where additional "keys" (for some reason), this would actually reset the `hasEvts` back to whatever value that was persisted (eg. immediate or some other later priority after "normal".

We should make this `hasEvts = hasEvts || !!evts.length;` so that once set to true its not reset to false while iterating over the for loop.